src/transformers/trainer.py

```diff
@@ -1837,6 +1837,8 @@ def _inner_training_loop(
             step = -1
             for step, inputs in enumerate(epoch_iterator):
                 total_batched_samples += 1
+                main_input_name = getattr(self.model, "main_input_name", "input_ids")
+                self.state.num_tokens_seen += self.accelerator.gather(inputs[main_input_name]).numel()
```

Comment:
Is this something we want to have by default? Firstly because of the additional computation we're adding but also the assumption that `inputs[main_input_name]` represents tokens.